### ISSUE TITLE
NullRef thrown by Internal.Cryptography.Pal.OpenSslX509ChainProcessor

### ISSUE THREAD

**Issue by StephenBonikowsky** [dotnet/corefx#37633]
In WCF we have [a test case](https://github.com/dotnet/wcf/blob/master/src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs#L424) that deliberately tries to use a service cert that is only valid for client authentication. We expect to get a 'SecurityNegotiationException'.

Starting recently we've been getting a NullRefException instead. This is only occurring on Linux (Centos.7) and I see that there has been some churn in 'System.Security.Cryptography.X509Certificates' under Pal.Unix and wondering if it could be related.

Here is the invalid certificate we used.
[InvalidEKUCert.zip](https://github.com/dotnet/corefx/files/3174762/InvalidEKUCert.zip)

```
System.NullReferenceException : Object reference not set to an instance of an object.
   at Internal.Cryptography.Pal.OpenSslX509ChainProcessor.AddUniqueStatus(IList`1 list, X509ChainStatus& status)
   at Internal.Cryptography.Pal.OpenSslX509ChainProcessor.ProcessPolicy(X509ChainElement[] elements, List`1 overallStatus, OidCollection applicationPolicy, OidCollection certificatePolicy)
   at Internal.Cryptography.Pal.OpenSslX509ChainProcessor.Finish(OidCollection applicationPolicy, OidCollection certificatePolicy)
   at Internal.Cryptography.Pal.ChainPal.BuildChain(Boolean useMachineContext, ICertificatePal cert, X509Certificate2Collection extraStore, OidCollection applicationPolicy, OidCollection certificatePolicy, X509RevocationMode revocationMode, X509RevocationFlag revocationFlag, DateTime verificationTime, TimeSpan timeout)
   at System.Security.Cryptography.X509Certificates.X509Chain.Build(X509Certificate2 certificate, Boolean throwOnException)
   at System.Security.Cryptography.X509Certificates.X509Chain.Build(X509Certificate2 certificate)
   at System.Net.Security.CertificateValidation.BuildChainAndVerifyProperties(X509Chain chain, X509Certificate2 remoteCertificate, Boolean checkCertName, String hostName)
   at System.Net.Security.SecureChannel.VerifyRemoteCertificate(RemoteCertValidationCallback remoteCertValidationCallback, ProtocolToken& alertToken)
   at System.Net.Security.SslStream.CompleteHandshake(ProtocolToken& alertToken)
```

Currently the "get repro environment" system via MissionControl is broken so I haven't been able to get a repro machine for this.
The test result in MC is [here](https://mc.dot.net/#/user/dotnet-mc-bot-2/official~2Fdotnet~2Fwcf~2Frefs~2Fheads~2Fmaster/build~2Ftest~2F/20190510.1/workItem/Client.ExpectedExceptions.IntegrationTests.dll/analysis/xunit/ExpectedExceptionTests~2FTCP_ServiceCertInvalidEKU_Throw_SecurityNegotiationException).

**Comment by StephenBonikowsky** [comment]
@bartonjs @stephentoub Do you guys know if any of the recent changes could have caused this?

**Comment by bartonjs** [comment]
Already fixed by #37318. (Assuming you're only seeing it in 3.0 preview builds)
